server/services/offenderService.ts

```diff
@@ -43,31 +43,32 @@ export default function createOffenderService(elite2ClientBuilder) {
     }
   }
 
-  const getOffenderImage = (token, bookingId) => {
+  const getOffenderImage = (token, bookingId): Promise<ReadableStream> => {
     const elite2Client = elite2ClientBuilder(token)
     return elite2Client.getOffenderImage(bookingId)
   }
 
-  const fullName = ({ firstName, lastName }) => `${properCaseName(lastName)}, ${properCaseName(firstName)}`
+  const fullName = ({ firstName, lastName }): string => `${properCaseName(lastName)}, ${properCaseName(firstName)}`
 
-  const getOffenderNames = async (token, offenderNos) => {
+  const getOffenderNames = async (token, offenderNos): Promise<{ [offenderNo: string]: string }> => {
     if (offenderNos.length === 0) {
       return {}
     }
     const uniqueNos = [...new Set(offenderNos)]
     const offenders = await elite2ClientBuilder(token).getOffenders(uniqueNos)
+
     return offenders.reduce((rv, offender) => ({ ...rv, [offender.offenderNo]: fullName(offender) }), {})
   }
 
-  const getLocation = (token, locationId) => {
+  const getLocation = (token: string, locationId: string): Promise<PrisonLocation> => {
     if (!locationId) {
-      return {}
+      return Promise.resolve({})
```

Comment:
meh.. would have thought that you could just return `{}` as returning from an async function will wrap the object in a Future.

Reply:
Oh yes!